.github/workflows/ci.yml

```diff
@@ -16,6 +16,11 @@ jobs:
         with:
           python-version: 3.x
 
+      - uses: actions/cache@v4
+        with:
+          path: ~/.cache/golangci-lint
+          key: pre-commit-3|${{ env.pythonLocation }}|${{ hashFiles('.pre-commit-config.yaml') }}
```

Comment:
Might not be the best idea to use the same cache key as pre-comit. Maybe you can do:


`pre-commit-golangci-lint|${{ env.pythonLocation }}|${{ hashFiles('.pre-commit-config.yaml') }}`

Reply:
According to the docs a cache can handle multiple directories [1]. Not sure though if using the cache action twice with different dirs on the same cache key gets us the same behavior. Would be interesting to test.

Our pre-commit-3 cache takes 170MB, while the pre-commit-golangci-lint cache has a size of 1.7MB. Might be more efficient to do a single cache restore with both dirs(?). Apparently its backbone is Azure blob storage [2].

I think we can leave it at using two caches, just found the topic interesting.

[1] https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/caching-dependencies-to-speed-up-workflows#input-parameters-for-the-cache-action
[2] https://depot.dev/blog/github-actions-cache

Reply:
I think we might want to have a different cache key. When using `pre-commit-golangci-lint|${{ env.pythonLocation }}|${{ hashFiles('.pre-commit-config.yaml') }}` the cache will never be updated until we either edit the pre-commit config or a new Python env is setup through the Python action.

We probably should include some hashes of our sources.

Reply:
If you want to be really precise, you might need to play with the `restore-key` action input, and use the input key you specified.